File: Labs/C++/Lab_15/Lab_15.cpp
// Lab 15
// Derek P Sifford
#include <iostream>
using namespace std;

void printArray(int[], int);
int secondSmallestItem(int[], int);

int main() {

    int arr[10]  = {14, 29, 44, 2, 400, 302, 39, 56, 18, 89};
    printArray(arr, 10);
    int secSmallest = secondSmallestItem(arr, 10);

    cout << "The second smallest item is " << secSmallest << endl;

    return 0;
}

void printArray(int arr[], int size) {
    for (int i = 0; i < size; i++) {
        cout << arr[i] << endl;
    }
}

int secondSmallestItem(int arr[], int size) {
    int firstSmallest = arr[0], secondSmallest = arr[0];

    for (int i = 0; i < size; i++) {
        if (arr[i] < firstSmallest) {
            secondSmallest = firstSmallest;
            firstSmallest = arr[i];
        }
    }

    return secondSmallest;

}

File: Labs/golang/Lab_15/Lab_15.go
package main

import (
	"fmt"
)

func main() {

	arr := [15]int{23, 3, 21, 5, 78, 52, 46, 73, 46, 73, 62, 77, 43, 48, 25}
	fmt.Println(arr)

	smallest := arr[0]
	var secondSmallest int

	for _, i := range arr {
		if i < smallest {
			smallest = i
		}
	}

	if smallest == arr[0] {
		secondSmallest = arr[1]
	} else {
		secondSmallest = arr[0]
	}

	for _, i := range arr {
		if i == smallest {
			continue
		}

		if i < secondSmallest {
			secondSmallest = i
		}
	}

	fmt.Println("The second smallest is ", secondSmallest)

}
